client/app/hosting/offer/upgrade/hosting-offer-upgrade.controller.js

```diff
@@ -15,113 +16,116 @@ angular.module('App').controller(
-      this.$scope.getDurations = () => this.getDurations();
-      this.$scope.loadContracts = () => this.loadContracts();
-      this.$scope.backToContracts = () => this.backToContracts();
-
       this.User.getUser()
         .then((user) => {
           this.ovhSubsidiary = user.ovhSubsidiary;
         });
 
-      this.Hosting.getAvailableOffer(_.get(this.hosting, 'serviceName', this.$stateParams.productId))
-        .then((availableOffer) => {
-          this.availableOffers = availableOffer;
+      this.Hosting.getSelected(this.productId)
+        .then((hosting) => {
+          this.hosting = hosting;
+          return this.Hosting.getAvailableOffer(this.productId);
+        })
+        .then((availableOffers) => {
+          this.availableOffers = availableOffers
+            .map(offer => ({
+              name: this.$translate.instant(`hosting_dashboard_service_offer_${offer}`),
+              value: offer,
+            }));
         })
         .catch(() => {
           this.availableOffers = [];
         })
         .finally(() => {
-          this.loading.availableOffer = false;
+          this.loading.availableOffers = false;
         });
     }
 
-    /* Step 1 */
-    orderByOffer(offer) {
-      return this.$translate.instant(`hosting_dashboard_service_offer_${offer}`) || offer;
-    }
-
-    /* Step 2 */
     getDurations() {
       this.durations = {
         available: [],
         details: {},
       };
       this.loading.durations = true;
 
-      return this.Hosting.getUpgradePrices(_.get(this.hosting, 'serviceName', this.$stateParams.productId), this.model.capacity)
+      return this.Hosting.getUpgradePrices(_.get(this.hosting, 'serviceName', this.$stateParams.productId), this.model.offer.value)
         .then((durations) => {
           this.durations.available = durations;
           if (durations.length === 1) {
             [this.model.duration] = this.durations.available;
           }
-        }, (err) => {
-          this.$scope.resetAction();
+        })
+        .catch((err) => {
           this.Alerter.alertFromSWS(this.$translate.instant('hosting_order_upgrade_error'), err, this.$scope.alerts.main);
-        }, (durations) => {
-          this.durations.available = durations;
         })
         .finally(() => {
           this.loading.durations = false;
         });
     }
 
-    /* Step 3 */
-    loadContracts() {
-      this.agree = false;
-      if (!this.model.duration.contracts || !this.model.duration.contracts.length) {
-        this.$rootScope.$broadcast('wizard-goToStep', 4);
-      }
-    }
+    formatTime(startTime) {
+      const time = moment(`2000-01-01T${startTime}`);
+      const now = moment(moment().format('2000-01-01THH:mm'));
+      const isToday = time.diff(now) > 0;
 
-    /* Step 4 */
-    backToContracts() {
-      if (!this.model.duration.contracts || !this.model.duration.contracts.length) {
-        this.$rootScope.$broadcast('wizard-goToStep', 3);
-      }
+      return isToday
```

Comment:
```suggestion
      return this.$translate.instant(
        isToday 
          ? 'hosting_order_upgrade_start_time_summary_today'
          : 'hosting_order_upgrade_start_time_summary_tomorrow', 
        { time: time.format('LT (UTCZ)') },
      )
```
would be nicer imo :)